### PR TITLE
Remove engine interference in microbenchmarks

### PR DESCRIPTION
**Pull request by Hixie**

### CODE REVIEW

File: packages/flutter_test/lib/src/binding.dart [hunk 1]
```diff
@@ -699,6 +699,16 @@ enum LiveTestWidgetsFlutterBindingFramePolicy {
...
   fullyLive,
+
+  /// Ignore any request to schedule a frame.
+  ///
+  /// This is intended to be used by benchmarks (hence the name) that drive the
+  /// pipeline directly. It tells the binding to entirely ignore requests for a
+  /// frame to be scheduled, while still allowing frames that are pumped
+  /// directly (invoking [Window.onBeginFrame] and [Window.onDrawFrame]) to run.
+  ///
+  /// Beware that [hasScheduledFrame] will never be true in this mode.
```
Comment (yjbanov) [inline packages/flutter_test/lib/src/binding.dart]: nit: "Beware that" seems to fall under the same category as "Note that".

File: packages/flutter_test/lib/src/binding.dart [hunk 2]
```diff
@@ -772,6 +782,12 @@ class LiveTestWidgetsFlutterBinding extends TestWidgetsFlutterBinding {
   ///   requests from the engine to be serviced, even those the test did not
   ///   explicitly pump.
   ///
+  /// * [LiveTestWidgetsFlutterBindingFramePolicy.benchmark] allows all frame
+  ///   requests from the engine to be serviced, and allows all frame requests
+  ///   that are artificially triggered to be serviced, but prevents the
+  ///   framework from requesting any frames from the engine itself. Beware that
```
Comment (yjbanov) [inline packages/flutter_test/lib/src/binding.dart]: ditto

**Review by yjbanov — APPROVED** [review]

**Comment by yjbanov** [comment]
lgtm

**Comment by Hixie** [comment]
Improved the documentation so that I don't need "beware that".

Thanks for the review. Will land on green.

**Pull request merged**